src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -3814,8 +3814,8 @@ private void ImportBinaryOperation(ILOpcode opcode)
                         break;
 
                     case ILOpcode.add_ovf:
-                        Debug.Assert(type.Category == TypeFlags.Int32 || type.Category == TypeFlags.Int64);
-                        if (type.Category == TypeFlags.Int32)
+                        Debug.Assert(type.Category == TypeFlags.Int32 || type.Category == TypeFlags.Int64 || type.Category == TypeFlags.Char);
```

Comment:
It does not sound right to have `char` on the IL stack. Should `char` be converted to `int32` when it gets loaded onto the IL stack instead?

Reply:
ILVerify does it in `StackValue.CreateFromType` method.

Reply:
Webassembly stack entries keep both the `StackValueKind` and the `TypeDesc` .  I've changed these asserts and ifs to use thte `StackValueKind` which I think is more correct and inline with `StackValue.CreateFromType`